conans/client/graph/proxy.py

```diff
@@ -59,67 +63,99 @@ def _get_recipe(self, reference, check_updates, update, remotes):
+            if latest_rrev:
+                # check if we already have the latest in local cache
+                # TODO: cache2.0 here if we already have a revision in the cache but we add the
+                #  --update argument and we find that same revision in server, we will not
+                #  download anything but we will UPDATE the date of that revision in the
+                #  local cache and WE ARE ALSO UPDATING THE REMOTE
+                #  Check if this is the flow we want to follow
+                cache_time = datetime.fromtimestamp(self._cache.get_timestamp(ref), gettz())
+                if latest_rrev.revision != ref.revision:
+                    if cache_time < remote_time:
+                        remotes.select(remote.name)
+                        # the remote one is newer
+                        output.info("Retrieving from remote '%s'..." % remote.name)
+                        remote, new_ref = self._download_recipe(latest_rrev, output,
+                                                                remotes, remote)
+                        new_recipe_layout = self._cache.ref_layout(new_ref)
+                        new_conanfile_path = new_recipe_layout.conanfile()
+                        status = RECIPE_UPDATED
+                        return new_conanfile_path, status, remote, new_ref
+                    else:
+                        status = RECIPE_NEWER
                 else:
-                    status = RECIPE_UPDATEABLE
+                    # TODO: cache2.0 we are returning RECIPE_UPDATED just because we are updating
+                    #  the date
+                    if cache_time >= remote_time:
+                        status = RECIPE_INCACHE
+                    else:
+                        selected_remote = remote
+                        remotes.select(remote.name)
+                        self._cache.update_reference(ref,
+                                                     new_timestamp=datetime.timestamp(remote_time),
+                                                     new_remote=selected_remote.name)
+                        status = RECIPE_INCACHE_DATE_UPDATED
+                return conanfile_path, status, selected_remote, ref
             else:
-                status = RECIPE_NEWER
+                status = RECIPE_NOT_IN_REMOTE
+                return conanfile_path, status, selected_remote, ref
         else:
             status = RECIPE_INCACHE
+            return conanfile_path, status, cur_remote, ref
+
+    def _get_rrev_from_remotes(self, reference, remotes, check_all_servers):
+        results = []
+        output = ScopedOutput(str(reference), self._out)
+
+        # TODO: cache2.0 --update strategies: when we have specified the revision we don't want to
+        #  check all the remotes, just return the first match
+        output.info(f"Checking all remotes: ({', '.join([remote.name for remote in remotes])})")
```

Comment:
I agree that this sounds as a good functional behavior, but I am also concerned about the performance penalty of resolving graphs when there are several (lets say 5, 10) remotes defined, as this multiplies xN the slowest part of the resolution, which is doing API calls to remote servers.

Not something to change here, but something to talk with @lasote and discuss.

Reply:
Yep, we might opt-in/out that behavior, you could always use Virtual repositories.

Reply:
note: we should benchmark virtual repos

Reply:
This has been finally changed, right @czoido? now it only do this for ``--update``?

Reply:
Yes, checking all the remotes will only happen with `--update`, probably this message that will only make sense with that argument is redundant or should be changed, I'll remove it for the moment.